modules/template/elements/BaseRecordElementVariable.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\custom_pages\modules\template\elements;
+
+use yii\db\ActiveRecord;
+
+class BaseRecordElementVariable extends BaseElementVariable
+{
+    protected ?ActiveRecord $record = null;
+
+    public function setRecord(?ActiveRecord $record): self
+    {
+        $this->record = $record;
+        return $this;
+    }
+
+    public function __isset($name): bool
```

Comment:
Not sure here, better introduce real properties.

Reply:
https://github.com/humhub/custom-pages-internal/issues/51#issuecomment-2970679907